cmd/libs/go2idl/client-gen/generators/generator_for_group.go

```diff
@@ -195,26 +191,20 @@ func (c *$.GroupVersion$Client) RESTClient() $.RESTClientInterface|raw$ {
 
 var newClientForRESTClientTemplate = `
 // New creates a new $.GroupVersion$Client for the given RESTClient.
-func New(c $.RESTClientInterface|raw$) *$.GroupVersion$Client {
-	return &$.GroupVersion$Client{c}
+func New(c $.RESTClientInterface|raw$, parameterCodec $.runtimeParameterCodec|raw$) *$.GroupVersion$Client {
```

Comment:
This is going to be a very unpopular change.  You should be able to keep the old API with a generated and working default param codec (which we'll use 99% of the time) and if you must, a separate method that can take it as a parameter.